clients/curl-client.rst

```diff
@@ -1,2 +1,76 @@
 cURL Client
 ===========
+
+This client uses `cURL PHP extension <http://php.net/curl>`_.
+
+Installation
+------------
+
+To install the cURL client, run:
+
+.. code-block:: bash
+
+    $ composer require php-http/curl-client
+
+Usage
+-----
+
+The cURL client needs a :ref:`message <message-factory>` and a :ref:`stream <message-factory>`
+factories in order to to work::
+
+    use Http\Client\Curl\Client;
+
+    $client = new Client($messageFactory, $streamFactory);
+
+Using `php-http/discovery <https://packagist.org/packages/php-http/discovery>`_::
+
+    use Http\Client\Curl\Client;
+    use Http\Discovery\MessageFactoryDiscovery;
+    use Http\Discovery\StreamFactoryDiscovery;
+
+    $messageFactory = MessageFactoryDiscovery::find();
+    $streamFactory = StreamFactoryDiscovery::find();
+    $client = new Client($messageFactory, $streamFactory);
+
+
+Using `mekras/httplug-diactoros-bridge <https://packagist.org/packages/mekras/httplug-diactoros-bridge>`_::
```

Comment:
Message factories are moved to php-http/message package. Doesn't it fulfill your requirements?